src/desktop.c

```diff
@@ -434,3 +434,45 @@ get_cursor_context(struct server *server)
 	wlr_log(WLR_ERROR, "Unknown node detected");
 	return ret;
 }
+
+void
+desktop_views_append(struct server *server, struct wl_array *views,
```

Comment:
Within the codebase, usually `desktop` refers to actions involving multiple views. `workspace` however refers to what most people know as "virtual desktop".

I agree though that the `_views_append()` part might be a bit too specific although it matches what the function does: adding view pointers to a supplied array. `desktop_append_views_to_array()` would be even more specific but should make clear what the function does. I think having the `append` part in there makes sense though, otherwise a dev might expect the function to initialize or clean the array before filling it (which we could actually do but it would restrict its uses in the future as well).

Reply:
The more I look at the code the more I suspect that it perhaps would be more natural do something like this:

```
view_for_each(view, &server->views, criteria) {
    [...]
}
```

That's a common pattern in the Linux kernel:

```
kvm_for_each_vcpu(i, vcpu, kvm) {}
for_each_online_cpu(cpu) {}
for_each_lpi_its {}
for_each_pcm_streams(stream) {}
```

Dunno, just my five cents :)

Reply:
That would indeed be a valid alternative approach, would involve macros though.

Reply:
Looks like they invoke `for_each_set_bit` in https://elixir.bootlin.com/linux/latest/source/tools/include/linux/bitops.h#L52
(although didn't look at all).

I prefer to approach we've taken. Avoids macro-bashing which I'm scared of.

Reply:
Alright, lets keep it at `wl_array` then.

Should we change the name of the function though?
I totally get the point that `desktop_views_append()` may sound like appending views to a desktop which is obviously not the case here. So far I still prefer `desktop_append_views_to_array()` the most but maybe there is a better self-explaining name.

Reply:
Yes, agree. `desktop_append_views_to_array()` is better.

We could consider putting the function in a dedicated translation unit and thereby drop the `desktop_`. How about `views.c` with `views_add()`

Reply:
Why not view.c?

Reply:
Also in our case no complex macro bashing required:

```
#define server_for_each_view(server, ptr, criteria) \
for (ptr = view_iterate(&server->views, ptr, criteria); ptr != NULL; ptr = view_iterate(ptr, ptr, criteria))
```

But I guess view rearrangement is really a strong argument for keeping things in array.